lib/livebook_web/live/hooks/sidebar_hook.ex

```diff
@@ -33,16 +39,13 @@ defmodule LivebookWeb.SidebarHook do
   defp handle_info(_event, socket), do: {:cont, socket}
 
   defp handle_event("shutdown", _params, socket) do
-    case Livebook.Config.shutdown_callback() do
-      {m, f, a} ->
-        apply(m, f, a)
-
-        {:halt, put_flash(socket, :info, "Livebook is shutting down. You can close this page.")}
-
-      _ ->
-        socket
-    end
+    Livebook.Config.shutdown()
+    shutdown(socket)
```

Comment:
```suggestion
    {:halt, socket}
```

Reply:
Nice! Hows does it work though? :D I thought returning :halt would terminate the genserver right away but it seems it would still handle_info the message from pubsub!

Reply:
`:halt` halts only the hook, so you don't fallback to the LiveView.